src/components/labels-layer/labels-layer.js

```diff
@@ -1,3 +1,78 @@
-import Component from './labels-layer-component';
+import { useEffect } from 'react';
+import { loadModules } from 'esri-loader';
+import { handleLayerCreation } from 'utils/layer-manager-utils';
+import { LABELS_LAYERS } from 'constants/layers-groups';
+import { layersConfig } from 'constants/mol-layers-configs';
+import { stylesConfig } from './labels-layer-styles-config';
 
-export default Component;
\ No newline at end of file
+const labelClassFactory = (LabelClassConstructor, styleGroup) => {
+  const config = stylesConfig[styleGroup];
+  return new LabelClassConstructor({
+    labelPlacement: 'above-center',
+    labelExpressionInfo: {
+      expression: "$feature.name"
+    },
+    where: `style = '${styleGroup}'`,
+    symbol: {
+      type: "label-3d",
+      symbolLayers: [{
+        type: "text",
+        font: {
+          family: config.fontFamily,
+          style: config.fontStyle,
+          size: config.fontSize,
+          weight: config.fontWeight || "normal"
+        },
+        material: { color: config.color },
+        halo: { size: 1 }
+      }]
+    }
+  });
+}
+
+const LabelsLayer = props => {
+  const { map } = props;
+
+  useEffect(() => {
+    loadModules(["esri/layers/support/LabelClass"])
+    .then(([labelClassConstructor]) => {
+      var style_1_ter = labelClassFactory(labelClassConstructor, 'style_1_ter');
```

Comment:
It's more like a suggestion, but we use here an array of slugs, like:
```suggestion
      const slugs = [ 'style_1_ter', 'style_2_ter', 'style_3_ter', ... ];
      const labelClasses = slugs.map(slug => {
        return labelClassFactory(labelClassConstructor, slug);
      });
      
      // so at the end:
      
      (...)
      .then(layer => {
            layer.opacity = 1;
            layer.visible = true;
            layer.labelsVisible = true;
            layer.labelingInfo = labelClasses
      })
      (...)
      
```

Reply:
Indeed 👏 
Nice one